import Service from "../models/serviseModel.js";

export const getAllServices = async (req, res) => {
    try {
        const services = await Service.find();
        res.status(200).json(services);
    } catch (error) {
        res.status(500).json({ error: 'Failed to fetch services', details: error.message });
    }
};

export const getServiceById = async (req, res) => {
    try {
        const service = await Service.findById(req.params.id);
        if (!service) {
            return res.status(404).json({ message: 'Service not found' });
        }
        res.status(200).json(service);
    } catch (error) {
        res.status(500).json({ error: 'Failed to fetch service', details: error.message });
    }
};

export const createService = async (req, res) => {
    try {
        const service = await Service.create(req.body);
        res.status(201).json(service);
    } catch (error) {
        res.status(500).json({ error: 'Failed to create service', details: error.message });
    }
};

export const updateService = async (req, res) => {
    try {
      const updatedService = await Service.findByIdAndUpdate(req.params.id, req.body, { new: true });
      if (!updatedService) {
        return res.status(404).json({ message: 'Service not found' });
      }
      res.status(200).json(updatedService);
    } catch (error) {
      res.status(500).json({ error: 'Failed to update service', details: error.message });
    }
  };
  

export const deleteService = async (req, res) => {
    try {
        const deletedService = await Service.findByIdAndDelete(req.params.id);
        if (!deletedService) {
            return res.status(404).json({ message: 'Service not found' });
        }
        res.status(200).json({ message: 'Service deleted successfully' });
    }
    catch (error) {
        res.status(500).json({ error: 'Failed to delete service', details: error.message });
    }
};